src/modules/chat/message/MessageContent.tsx

```diff
@@ -0,0 +1,126 @@
+/**
+ * Copyright 2024 IBM Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { AssistantPlan } from '@/app/api/threads-runs/types';
+import { Spinner } from '@/components/Spinner/Spinner';
+import throttle from 'lodash/throttle';
+import { useEffect, useMemo, useState } from 'react';
+import { getLastCompletedStep } from '../assistant-plan/utils';
+import { useRunContext } from '../providers/RunProvider';
+import { ChatMessage } from '../types';
+import { MarkdownContent } from './MarkdownContent';
+import classes from './Message.module.scss';
```

Comment:
We should probably also separate the CSS file. It's not recommended to import same CSS module into several different components in Next.